calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
Having `calico/node` do the auto-detection is very useful for cases just like this.  I think we will rely on autodetection for the DaemonSet deployment.

I looked at `valueFrom:fieldRef:fieldPath:status.podIP`, but the value there seems less reliable than our auto-detection and we don't have control over it.  Could be that we patch k8s to get a better value here, or get the `nodeIP` to be available in the downward API (but that might never happen because the k8s devs don't like the idea too much).